signature.go

```diff
@@ -133,16 +133,30 @@ func (s *Signature) sign(key string, r *http.Request) error {
 
 // IsValid validates this signature for the given key
 func (s Signature) IsValid(key string, r *http.Request) bool {
+	return s.isValid(key, r)
+}
+
+// IsValidRSA validates that the signature was signed by an RSA private key.
+func (s Signature) IsValidRSA(key *rsa.PublicKey, r *http.Request) bool {
```

Comment:
One thing to take note of here. I didn't want to break the existing api for `IsValid`, but I also don't want this code to be dealing with RSA keys as strings, and would rather just have consumers parse the public/private keys.

So, I just added the the `IsValidRSA`/`SignRequestRSA` versions of the `IsValid`/`SignRequest` methods. The downside is that this leaks details about the underlying algorithm. Alternatively, I could add a more generic `IsValid<xxx>(key interface{}, r *http.Request)` method as well so that you can pass a string for HMAC, or an `*rsa.PublicKey` for RSA.

Reply:
Note to self. Also need an `AuthRequestRSA` method.